pandas_schema/validation.py

```diff
@@ -84,7 +84,7 @@ def get_errors(self, series: pd.Series, column: 'column.Column'):
         simple_validation = ~self.validate(series)
         if column.allow_empty:
             # Failing results are those that are not empty, and fail the validation
-            validated = (series.str.len() > 0) & simple_validation
+            validated = ((series.dtype == 'str' and series.str.len() > 0) | (series.dtype == float and not series.isna)) & simple_validation
```

Comment:
I think this would be clearer and probably more efficient if this were a series of if statements rather than one huge expression